control-plane-operator/controllers/hostedcontrolplane/routecm/config.go

```diff
@@ -0,0 +1,81 @@
+package routecm
+
+import (
+	"fmt"
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/kas"
+	"path"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	openshiftcpv1 "github.com/openshift/api/openshiftcontrolplane/v1"
+
+	"github.com/openshift/hypershift/control-plane-operator/controllers/hostedcontrolplane/common"
+	"github.com/openshift/hypershift/support/api"
+	"github.com/openshift/hypershift/support/certs"
+	"github.com/openshift/hypershift/support/config"
+	"github.com/openshift/hypershift/support/util"
+)
+
+const (
+	configKey = "config.yaml"
+)
+
+func ReconcileOpenShiftRouteControllerManagerConfig(cm *corev1.ConfigMap, ownerRef config.OwnerRef, minTLSVersion string, cipherSuites []string, networkConfig *configv1.NetworkSpec) error {
+	ownerRef.ApplyTo(cm)
+
+	if cm.Data == nil {
+		cm.Data = map[string]string{}
+	}
+	config := &openshiftcpv1.OpenShiftControllerManagerConfig{}
```

Comment:
why do we need to deserialize this instead of just construct the struct, serialize and store it?

Reply:
@enxebre I thought it was to potentially allow changes to the config by another actor (similar to OCM). But I can remove the deserialization part if that is not needed.